crates/rust-analyzer/src/handlers.rs

```diff
@@ -568,15 +571,86 @@ pub(crate) fn handle_completion(
     };
     let line_index = snap.analysis.file_line_index(position.file_id)?;
     let line_endings = snap.file_line_endings(position.file_id);
+    let mut completion_resolve_data = FxHashMap::default();
+
     let items: Vec<CompletionItem> = items
         .into_iter()
-        .flat_map(|item| to_proto::completion_item(&line_index, line_endings, item))
+        .enumerate()
+        .flat_map(|(item_index, item)| {
+            let mut new_completion_items =
+                to_proto::completion_item(&line_index, line_endings, item.clone());
+
+            if snap.config.completion.resolve_additional_edits_lazily() {
+                if let Some(import_edit) = item.import_to_add() {
+                    completion_resolve_data.insert(
+                        item_index,
+                        CompletionResolveData {
+                            file_id: position.file_id,
```

Comment:
I think in this case it may make sense to just store the `params.text_document_position` instead of a `FileId` and reconstruct it during resolve. That will also get around any weird changes that may occur between requests. There's also nothing wrong with stuffing the params into `data` if that helps.

Reply:
Yes, I've noticed how other handlers use to parse the document uri to get the `FileId`, I've used this approach.